ergotree-interpreter/src/eval/block.rs

```diff
@@ -11,14 +12,31 @@ use crate::eval::Evaluable;
 
 impl Evaluable for BlockValue {
     fn eval(&self, env: &mut Env, ctx: &mut EvalContext) -> Result<Value, EvalError> {
+        let mut existing_variables = HashMap::new();
```

Comment:
This code breaks the `diag_on_reduced_to_false` test. We might need to make an exception for the global `BlockValue` of a script, and do not delete introduced `ValId`s?

Reply:
Thank you for taking care of this! Ideally, yes. If not, feel free to make the test ignored/commented.

Reply:
No problem! Made the change so `ValDef`s in global block persist.